app.js

```diff
@@ -0,0 +1,207 @@
+// this is the game board
+const gameBoard = () => {
+
+    const _board = [['','',''],['','',''],['','','']];
+
+    const setMark = (mark,i,j) => {
+      if(_board[i][j].length > 0){
+        return false;
+      }else{
+        _board[i][j] = mark;
+        return true;
+      }
+    }
+
+const checkIfFull = () => {
+          let isFull = true;
+          _board.forEach(row => {
+            row.forEach(cell => {
+                if(cell.length == 0){
+                  isFull = false;
+                }
+            })
+          })
+          return isFull;
+        }
+
+    const getBoard = () => _board;
+
+    const checkForWinner = (mark)=> {
+
+      // horizontal
+      if(_board[0][0] == mark && _board[0][1] == mark && _board[0][2] == mark){ return true; }
+      if(_board[1][0] == mark && _board[1][1] == mark && _board[1][2] == mark){ return true; }
+      if(_board[2][0] == mark && _board[2][1] == mark && _board[2][2] == mark){ return true; }
+
+      // vertical
+      if(_board[0][0] == mark && _board[1][0] == mark && _board[2][0] == mark){ return true; }
+      if(_board[0][1] == mark && _board[1][1] == mark && _board[2][1] == mark){ return true; }
+      if(_board[0][2] == mark && _board[1][2] == mark && _board[2][2] == mark){ return true; }
+
+      // diagonal
+      if(_board[0][0] == mark && _board[1][1] == mark && _board[2][2] == mark){ return true; }
+      if(_board[0][2] == mark && _board[1][1] == mark && _board[2][0] == mark){ return true; }
+
+      return false;
+    }
+
+    return { getBoard, setMark, checkIfFull,checkForWinner };
+}
+const playerFactory = (name, mark) => {
+    const getName = () => name;
```

Comment:
Why does `playerFactory` need to have the ff: `getName`, `getMark`, `setName` ?

Reply:
I still don't think it is necessary to have a `getName`, `getMark`and `setName` methods. Kindly clarify why we need them pls?

Reply:
It was a misinformation on out part, we had some errors and we figured we needed getters and setters.